web/src/hooks/useTodos.tsx

```diff
@@ -59,17 +55,12 @@ const useTodos = (): {
 
   const toggleItem = (id: string) => {
     setLoading(true)
-    const index: number = todos.findIndex(
-      (item: AddTodoResponse) => item.id === id
-    )
+    const index: number = todos.findIndex((item: TodoItem) => item.id === id)
     const todoItem = todos[index]
     todoAPI
-      .updateById({
-        id: id,
-        updateTodoRequest: {
-          is_completed: !todoItem.is_completed,
-          title: todoItem.title,
-        },
+      .updateById(id, {
+        is_completed: !todoItem.is_completed,
```

Comment:
⚠️ `is_completed` is optional, meaning you are "lying" to the compiler here by telling it that the property cannot be null or undefined.

Reply:
I might be misunderstanding you, but how is he "lying"? My first assumption is that you're referring to the use of `!` here. In this case, the `!` does not refer to the [non-null assertion operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator), but simply the negating operator (`!` = `not`).

Reply:
You're right! 😅 Well, as I said I'm struggling to focus and keep up with this now haha 😵‍💫 Good that you're still going strong!!